graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -523,6 +527,76 @@ def prune_output_blocks_in_compound_match_query(compound_match_query):
         return CompoundMatchQuery(match_queries=match_queries)
 
 
+def collect_filters_to_first_location_instance(compound_match_query):
+    """Collapse all filters applied to a particular location into a single `as_block`.
+
+    Specifically, for any location `L`, find all filters present on `L`, and apply
+    the conjunction of all these filters to the first occurence of `L` (within its MatchQuery).
+    """
+    def predicate_list_to_where_block(predicate_list):
```

Comment:
let's avoid having nested functions in general whenever possible, and use private (underscored) helper functions instead